Automattic/newspack-plugin
# feat: use reCAPTCHA to secure all Reader Activation-related forms

### All Submissions:






### Changes proposed in this Pull Request:



Implements reCAPTCHA v3 verification for all Reader Activation-related forms: registration, auth, subscribe (requires https://github.com/Automattic/newspack-newsletters/pull/935), and donation (requires https://github.com/Automattic/newspack-blocks/pull/1244).

As part of this PR, the wizard UI to enable reCAPTCHA and enter credentials have been moved from the **Reader Revenue > Stripe Settings** wizard to the **Engagement > Reader Activation** wizard. However, because reCAPTCHA is also used by the subscribe and streamlined donation blocks whether or not Reader Activation is enabled on the site, the reCAPTCHA script and methods can be enqueued and used even if Reader Activation features are not enabled.

<img width="1046" alt="Screen Shot 2022-08-16 at 6 46 28 PM" src="https://user-images.githubusercontent.com/2230142/185010642-c191dcc3-999c-48f9-9c33-4fb08119ce3e.png">

Also note that the UI for the reCAPTCHA settings may look out of place right now, but are styled to fit in with [this pending PR](https://github.com/Automattic/newspack-plugin/pull/1852).

Also implements the correct capitalization of reCAPTCHA in all UI.

### How to test the changes in this Pull Request:

1. On `master` for this plugin, Newsletters, and Blocks, confirm that reCAPTCHA is set up and working on your test site by following testing steps 3–7 from [this PR](https://github.com/Automattic/newspack-blocks/pull/1234).
2. Check out this branch, (https://github.com/Automattic/newspack-newsletters/pull/935) and https://github.com/Automattic/newspack-blocks/pull/1244. Confirm that the UI to enable reCAPTCHA and enter creds and all related values are automatically migrated from **Reader Revenue > Stripe Settings** to **Engagement > Reader Activation**.
3. Confirm that reCAPTCHA still works on streamlined donate blocks by setting the reCAPTCHA threshold (now on [this line](https://github.com/Automattic/newspack-plugin/compare/feat/recaptcha-for-all-forms?expand=1#diff-3ce140c40dae980330bca82b55eaf67a4f7abbbcc4f6609368fcf55ea696a7a7R24)) to `1.0` and trying to submit a donation. You should see an error message stating `User action failed captcha challenge`.
4. With the threshold still at `1.0`, confirm that reCAPTCHA now also works on login, subscribe, and register blocks.
5. Disable Reader Activation (either via the wizard option or by removing the environment variable from `wp-config.php`) and confirm that reCAPTCHA still works for streamlined donate and subscribe blocks, both of which don't require Reader Activation in order to function.
6. Re-enable Reader Activation, but disable reCAPTCHA via the wizard UI. Confirm that donation, login, subscribe, and register forms all work again without reCAPTCHA.

### Other information:
